### PR TITLE
[Enhancement] Run lake table publish version tasks in thread pool to avoid blocking bthread workers

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceProxy.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceProxy.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceProxy.java|1|1|100.00%|

File: be/src/service/service_be/lake_service.cpp [hunk 1]
```diff
@@ -114,20 +197,22 @@ void LakeServiceImpl::publish_version(::google::protobuf::RpcController* control
...
             response->add_failed_tablets(tablet_id);
         }
     }
+
+#ifdef BE_TEST
+    context->wait();
+#endif
 }
 
 void LakeServiceImpl::abort_txn(::google::protobuf::RpcController* controller,
```
Comment (wuleistarrocks) [inline be/src/service/service_be/lake_service.cpp:218]: abort_txn will delete_txn_log internal, which maybe time consuming for S3 storage, so the abort_txn thread may get blocked.

Comment (sduzh) [reply be/src/service/service_be/lake_service.cpp:218]: Yes you're right, I will submit another PR to fix this.

File: be/src/service/service_be/lake_service.cpp [hunk 2]
```diff
@@ -90,10 +97,86 @@ Status LakeServiceImpl::publish(lake::Tablet* tablet, const ::starrocks::lake::P
...
+#endif
+    // Will call `_context->done->Run()` if this is the ref count of _context is 1.
+    _context.reset();
+}
+
 void LakeServiceImpl::publish_version(::google::protobuf::RpcController* controller,
                                       const ::starrocks::lake::PublishVersionRequest* request,
                                       ::starrocks::lake::PublishVersionResponse* response,
                                       ::google::protobuf::Closure* done) {
+    LOG(INFO) << "Publishing version";
```
[the remark this answers is not in the document]
Comment (sduzh) [reply be/src/service/service_be/lake_service.cpp]: Removed

**Review by wuleistarrocks — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceProxy.java|1|1|100.00%|

**Review by imay — APPROVED** [review]

**Pull request merged**